tests/test_src/test_provider/test_proxyprovider.cpp

```diff
@@ -112,4 +112,36 @@ TEST(ProxyProvider, getTransactionStatus_invalidHash)
                      }
                  }, std::runtime_error );
 }
+
+TEST(ProxyProvider, getESDTokenBalance)
+{
+    ProxyProvider proxy("https://testnet-gateway.elrond.com");
+    Address const address("erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l");
+    auto const balance = proxy.getESDTTokenBalance(address, "this esdt does not exist");
+
+    EXPECT_TRUE(balance == DEFAULT_BALANCE);
+}
+
+TEST(ProxyProvider, getAllESDTokenBalances_noTokens)
+{
+    ProxyProvider proxy("https://testnet-gateway.elrond.com");
+    Address const address("erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l");
+    auto const esdts = proxy.getAllESDTTokenBalances(address);
+
+    EXPECT_TRUE(esdts.empty());
+}
+
+TEST(ProxyProvider, getAllESDTokenBalances_multipleTokens)
+{
+    ProxyProvider proxy("https://testnet-gateway.elrond.com");
+    Address const address("erd1nqtv8gdsf55xj9eg0wc8ar6ml46kpld2c86aq670mxgcf49sduzqaeyngx");
+    auto const esdts = proxy.getAllESDTTokenBalances(address);
+
+    EXPECT_TRUE(esdts.find("00040-4c4d18") != esdts.end());
```

Comment:
Tricky test. I don't think it will pass after a testnet reset or when `erd1nqtv...` transfers its tokens.

Perhaps (in a future PR), an option would be to make the integration tests depend on a local testnet (erdpy), and also add some setup / initialization steps before running the actual tests (e.g. steps such as creating some ESDT tokens etc.).

Reply:
Totally agree, these tests are not safe. That's also the reason why these tests are deactivated by `HTTP_PRECONDITIONS` flag. Your suggestion is very good. Also had a discussion with @AdoAdoAdo and we should use a local testnet in the future.